File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    
    lst = lost.copy()
    lost.sort()
    reserve.sort()
    
    for c in lost:
        if c in reserve:            # 도난과 빌려줄 수 있는 학생이 같을 때
            lst.remove(c)
            reserve.remove(c)       
        elif c - 1 in reserve and c - 1 not in lost:      # 앞 번호 학생에게 빌릴 수 있을 때
            lst.remove(c)
            reserve.remove(c - 1)
        elif c + 1 in reserve and c + 1 not in lost:      # 뒷 번호 학생에게 빌릴 수 있을 때
            lst.remove(c)
            reserve.remove(c + 1)
    
    return n - len(lst)

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {    
    // var lst = [...lost]
    
    lost.sort((a, b) => a - b)
    reserve.sort((a, b) => a - b)
    
    lost.map((num) => {
        if (reserve.includes(num)) {
            reserve = reserve.filter(e => e != num)
            lost = lost.filter(e => e != num)
        } else if (reserve.includes(num - 1) && !lost.includes(num - 1)) {
            reserve = reserve.filter(e => e != num - 1)
            lost = lost.filter(e => e != num)  
        } else if (reserve.includes(num + 1) && !lost.includes(num + 1)) {
            reserve = reserve.filter(e => e != num + 1)
            lost = lost.filter(e => e != num)
        }
    })
    
    return n - lost.length;
}